internal/controller/pod_controller.go

```diff
@@ -78,15 +87,58 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 	}
 	reqImagesHash := annotations[constants.CatGateImagesHashAnnotation]
 
+	// prevents removing the scheduling gate based on information before the cache is updated.
+	if value, ok := GateRemovalHistories.Load(reqImagesHash); ok {
+		lastGateRemovalTime := time.UnixMilli(value.(int64))
+		if time.Since(lastGateRemovalTime) < time.Millisecond*time.Duration(gateRemovalDelayMilliSecond) {
```

Comment:
Could you write the duration as follows? I think it is not general to write the units first.
`<meaningful value>*<unit>`

```suggestion
		if time.Since(lastGateRemovalTime) < time.Duration(gateRemovalDelayMilliSecond)*time.Millisecond {
```

Reply:
I fixed it with 85abd7d6096ef9c7245086f2b14da1f546f166bb.

Reply:
Could you fix other parts?